src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +32,75 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfig(remoteCert *x509.Certificate) (*tls.Config, error) {
+	if remoteCert == nil {
+		return nil, fmt.Errorf("invalid remote public key")
+	}
+
+	config := &tls.Config{}
+
+	// Add the public key to the CA pool to make it trusted.
+	config.RootCAs = x509.NewCertPool()
+	remoteCert.IsCA = true
+	remoteCert.KeyUsage = x509.KeyUsageCertSign
+	config.RootCAs.AddCert(remoteCert)
+
+	// Always use public key DNS name rather than server cert, so that it matches.
+	if len(remoteCert.DNSNames) > 0 {
+		config.ServerName = remoteCert.DNSNames[0]
+	}
+
+	return config, nil
+}
+
+func CreateHTTPClientWithCert(cert *x509.Certificate) (*http.Client, error) {
+	// Create the client
+	tlsConfig, err := TLSClientConfig(cert)
+	if err != nil {
+		return nil, err
+	}
+
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+	}, nil
+}
+
+func GetRemoteCertificate(address string) (*x509.Certificate, error) {
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	url, err := url.Parse("https://" + address)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse address: %w", err)
+	}
+
+	// Connect
+	req, err := http.NewRequest("GET", url.String(), nil)
```

Comment:
Is this to validate the address? That could work _by accident_, but the error message would still be weird as it will be descibing an error with a URL, and the user does not know what to do.

Instead, use `net.SplitHostPort` to verify the `address`, and then you can do `fmt.Sprintf("https://%s", address)` on the HTTP request